helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -646,10 +645,14 @@ private void subscribeForChanges(NotificationContext.Type callbackType, String p
               break;
             }
             default: {
-              List<String> childNames = _zkClient.getChildren(path);
-              if (childNames != null) {
-                for (String childName : childNames) {
-                  String childPath = path + "/" + childName;
+              // When callback type is FINALIZE or PERIODIC_REFRESH, subscribeChildChange doesn't
+              // get children list. We need to read children to unsubscribe data change listeners.
+              if (callbackType != Type.INIT && callbackType != Type.CALLBACK) {
```

Comment:
As I mentioned above, if the return value is ChildrenSubscribeResult, then you just need to check "_isInstalled" for the following logic.

Reply:
Replied above.